draft-ietf-ppm-dap.md

```diff
@@ -1305,50 +1493,89 @@ given batch interval, denoted `batch_interval`, decryption works as follows:
-`batch_interval.start` and `batch_interval.duration` are divisible by
-`min_batch_duration` and that `batch_interval.duration >= min_batch_duration`.
-Unless both these conditions are true, the aggregator MUST abort and alert its
-peer with error "batchInvalid".
-
-Next, the aggregator checks that the request respects the generic privacy
-parameters of the DAP task. Let `X` denote the set of reports for which the
-aggregator has recovered a valid output share and which fall in the batch
-interval of the request.
-
-* If `len(X) < min_batch_size`, then the aggregator MUST abort and alert its
-  peer with "insufficientBatchSize".
-* The aggregator keeps track of the number of times each report was added to the
-  batch of an AggregateShareReq. If any report in `X` was added to at least
-  `max_batch_lifetime` previous batches, then the aggregator MUST abort and
-  alert the peer with "batchLifetimeExceeded".
-
-Finally, the aggregator checks that the batch interval defined by the collect
-request satisfies one of the conditions:
-
-1. The batch interval does not overlap with the batch interval of any prior
-   completed collect requests.
-1. The batch interval, including its start and duration values, match a prior
-   completed collect request.
+First the aggregator checks that the request respects any batch "boundaries"
+determined by the query configuration:
+
+* For time-interval tasks, the query configuration specifies the "minimum batch
+  duration", which is determined by `time_precision`. For the `batch_interval`
+  included with the query, the Aggregator checks that both
+  `batch_interval.start` and `batch_interval.duration` are divisible by
+  `time_precision` and that `batch_interval.duration >= time_precision`.
+
+* For fixed-size tasks, the batch is assocaited with a "batch ID" selected by
+  the Leader. Thus the Aggregator needs to check that the query is associated
+  with a known batch ID:
+
+    * If the query contained in the CollectReq has type `by-batch-id`, the
+      Leader checks that the provided batch ID corresponds to a batch ID it
+      returned in a previous CollectResp for the task.
+
+    * The Helper checks that the batch ID provided by the leader in its
+      AggregateShareReq corresponds to a batch ID used in a previous
+      AggregateInitializeReq for the task.
+
+If the boundary check fails, then the Aggregator MUST abort with error
+"batchInvalid".
+
+Next, the Aggregator checks that batch contains a valid number of reports. Let
+`X` denote the set of reports in the batch:
+
+* For time-interval tasks, the query configuration specifies the minimum batch
+  size, `min_batch_size`. The Aggregator checks that `len(X) >= min_batch_size`.
+
+* For fixed-size tasks, the query configuration specifies the minimum batch
+  size, `min_batch_size`, and maximum batch size. The Aggregator checks that
+  `len(X) >= min_batch_size` and `len(X) < max_batch_size`.
```

Comment:
nit: `max_batch_size` is the only upper bound that's exclusive, should we make it `<=` ? @cjpatton

Reply:
Done.